File: src/2025-02-22-00-51-50--a-deepseek-r1-test-220-10-1--2-1-huoshan/go/exercises/practice/say/say.go
package say

import (
	"strings"
)

var ones = [...]string{
	"zero", "one", "two", "three", "four",
	"five", "six", "seven", "eight", "nine",
	"ten", "eleven", "twelve", "thirteen", "fourteen",
	"fifteen", "sixteen", "seventeen", "eighteen", "nineteen",
}

var tens = [...]string{
	"", "", "twenty", "thirty", "forty",
	"fifty", "sixty", "seventy", "eighty", "ninety",
}

var scales = [...]string{
	"", "thousand", "million", "billion",
}

func Say(n int64) (string, bool) {
	if n < 0 || n > 999_999_999_999 {
		return "", false
	}

	if n == 0 {
		return ones[0], true
	}

	var chunks []int64
	for n > 0 {
		chunks = append(chunks, n%1000)
		n /= 1000
	}

	var parts []string
	for i, chunk := range chunks {
		if chunk == 0 {
			continue
		}
		part := convertChunk(chunk)
		if i > 0 {
			part += " " + scales[i]
		}
		parts = append([]string{part}, parts...)
	}

	return strings.Join(parts, " "), true
}

func convertChunk(n int64) string {
	var parts []string
	
	if n >= 100 {
		parts = append(parts, ones[n/100]+" hundred")
		n %= 100
	}

	if n >= 20 {
		part := tens[n/10]
		if n%10 != 0 {
			part += "-" + ones[n%10]
		}
		parts = append(parts, part)
	} else if n > 0 {
		parts = append(parts, ones[n])
	}

	return strings.Join(parts, " ")
}

File: src/2025-02-22-00-51-50--a-deepseek-r1-test-220-10-1--2-1-huoshan/javascript/exercises/practice/say/say.js
//
// This is only a SKELETON file for the 'Say' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const say = (n) => {
  if (n < 0 || n > 999999999999) {
    throw new Error('Number must be between 0 and 999,999,999,999.');
  }

  if (n === 0) return 'zero';

  const numbers0to19 = ['', 'one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine',
                      'ten', 'eleven', 'twelve', 'thirteen', 'fourteen', 'fifteen', 'sixteen',
                      'seventeen', 'eighteen', 'nineteen'];
  const tens = ['twenty', 'thirty', 'forty', 'fifty', 'sixty', 'seventy', 'eighty', 'ninety'];

  const processChunk = (num) => {
    if (num === 0) return '';
    const parts = [];
    const hundreds = Math.floor(num / 100);
    const remainder = num % 100;
    
    if (hundreds > 0) {
      parts.push(`${numbers0to19[hundreds]} hundred`);
    }
    
    if (remainder > 0) {
      if (remainder < 20) {
        parts.push(numbers0to19[remainder]);
      } else {
        const ten = Math.floor(remainder / 10);
        const unit = remainder % 10;
        parts.push(unit === 0 ? tens[ten - 2] : `${tens[ten - 2]}-${numbers0to19[unit]}`);
      }
    }
    
    return parts.join(' ');
  };

  const splitNumber = (num) => {
    const chunks = [];
    while (num > 0) {
      chunks.push(num % 1000);
      num = Math.floor(num / 1000);
    }
    return chunks;
  };

  const chunks = splitNumber(n);
  const scales = ['', 'thousand', 'million', 'billion'];
  const resultParts = [];
  
  chunks.forEach((chunk, index) => {
    const chunkText = processChunk(chunk);
    if (chunkText) {
      const scale = scales[index];
      resultParts.unshift(`${chunkText}${scale ? ' ' + scale : ''}`);
    }
  });

  return resultParts.join(' ');
};
